pkg/controllers/management/authprovisioningv2/cluster_test.go

```diff
@@ -75,25 +75,36 @@ func TestOnCluster(t *testing.T) {
 		prtbMock      func() mgmtcontrollers.ProjectRoleTemplateBindingController
 		expectedErr   error
 	}{
-		"no rke doesn't enqueue CRTBs": {
+		"no rke enqueue CRTBs and PRTBs": {
```

Comment:
```suggestion
		"non-rke enqueue CRTBs and PRTBs": {
```

But given the nature of this change, maybe this and the next test can be folded into one as it's expected to be the same behaviour across all providers?

Reply:
Fair; I've given that a try in an additional commit, curious to know how close that might be to what you had in mind ;)

Reply:
There's now only one code-path in `enqueueRoleTemplateBindings`.

There are two code paths outside, one where the the role exists, and the other where it doesn't?

The behaviour doesn't change based on the cluster-type?

Reply:
Right, I've refactored test cases to make the distinction between those 2 code paths.